src/main/webapp/app/exercises/shared/participation-submission/participation-submission.component.ts

```diff
@@ -72,4 +129,52 @@ export class ParticipationSubmissionComponent implements OnInit {
                 }
             });
     }
+
+    getName() {
+        if (this.participation?.type === ParticipationType.STUDENT) {
+            return (this.participation as StudentParticipation).student?.name || (this.participation as StudentParticipation).team?.name;
+        } else if (this.participation?.type === ParticipationType.SOLUTION) {
+            return this.translate.instant('artemisApp.participation.solutionParticipation');
+        } else if (this.participation?.type === ParticipationType.TEMPLATE) {
+            return this.translate.instant('artemisApp.participation.templateParticipation');
+        }
+        return 'N/A';
+    }
+
+    isTemplateOrSolutionParticipation(): boolean {
+        return this.participation?.type === ParticipationType.TEMPLATE || this.participation?.type === ParticipationType.SOLUTION;
+    }
+
+    getCommitUrl(submission: ProgrammingSubmission): string | undefined {
+        let repoUrl: string | undefined;
+        if (submission?.type === SubmissionType.TEST) {
+            repoUrl = (this.exercise as ProgrammingExercise).testRepositoryUrl;
+        } else if (this.participation?.type === ParticipationType.PROGRAMMING) {
+            repoUrl = (this.participation as ProgrammingExerciseStudentParticipation).repositoryUrl;
+        } else if (this.participation?.type === ParticipationType.SOLUTION) {
+            repoUrl = (this.participation as SolutionProgrammingExerciseParticipation).repositoryUrl;
+        } else if (this.participation?.type === ParticipationType.TEMPLATE) {
+            repoUrl = (this.participation as TemplateProgrammingExerciseParticipation).repositoryUrl;
+        }
+        if (repoUrl) {
+            if (this.activeProfiles) {
+                // Remove ".git" suffix
+                const baseUrl = repoUrl.replace('.git', '');
+                if (this.activeProfiles.includes('bitbucket')) {
```

Comment:
I don't really like having a dependency on bitbucket and gitlab here. We should try to solve this differently. I guess the server could create a template url (based on the configuration whether we use Gitlab or Jenkins) and send this in the profile info call. The client then only needs to replace <commit-hash> with the actual value.

Reply:
this is fixed in the latest commits